src/main/java/com/metamx/emitter/core/HttpPostEmitter.java

```diff
@@ -125,39 +131,57 @@ public HttpPostEmitter(
     catch (MalformedURLException e) {
       throw new ISE(e, "Bad URL: %s", config.getRecipientBaseUrl());
     }
+    concurrentBatch.set(new Batch(this, acquireBuffer(), 0));
   }
 
   @Override
   @LifecycleStart
   public void start()
   {
-    synchronized (started) {
-      if (!started.getAndSet(true)) {
-        exec.schedule(
-            new EmittingRunnable(version.get()),
-            config.getFlushMillis(),
-            TimeUnit.MILLISECONDS
-        );
+    synchronized (startLock) {
+      if (!started) {
```

Comment:
startLatch + started encode three states:
startLatch.count = 1 and started = false: not started
startLatch.count = 0 and started = true: started, running
startLatch.count = 0 and started = false: finished.

This is a limited replacement for LifecycleLock: https://github.com/druid-io/druid/pull/3964, it is committed to Druid so couldn't be used in emitter.